src/XPurchase/VirtoCommerce.XPurchase/CartAggregate.cs

```diff
@@ -156,13 +156,19 @@ public virtual async Task<CartAggregate> AddItemAsync(NewCartItem newCartItem)
             {
                 throw new ArgumentNullException(nameof(newCartItem));
             }
-
-            var validationResult = await AbstractTypeFactory<NewCartItemValidator>.TryCreateInstance().ValidateAsync(newCartItem, options => options.IncludeRuleSets(ValidationRuleSet));
-            if (!validationResult.IsValid)
-            {
-                ValidationErrors.AddRange(validationResult.Errors);
-            }
-            else if (newCartItem.CartProduct != null)
+
+            if (Cart.Type != "Wishlist")
+            {
+                var validationResult = await AbstractTypeFactory<NewCartItemValidator>.TryCreateInstance().ValidateAsync(newCartItem, options => options.IncludeRuleSets(ValidationRuleSet));
+                if (!validationResult.IsValid)
+                {
+                    ValidationErrors.AddRange(validationResult.Errors);
+                    return this;
+                }
+
+            }
+
+            if (newCartItem.CartProduct != null)
```

Comment:
Undo. 
Use `cartAggregate.ValidationRuleSet = new string[] { "default" };` after getting cartAggregate in `AddWishlistItemCommandHandler`